File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        char[] romanArr= s.toCharArray();
        int len=romanArr.length;
        int FinalValue = 0;
        int curVal, nextVal;
        for(int i=0; i<len; i++){
            curVal = value(romanArr[i]);
            if(i!=len-1){
                nextVal = value(romanArr[i+1]);
            }else{
                nextVal = 0;
            }
            if(curVal>=nextVal){
                FinalValue = FinalValue + curVal;
            }else{
                FinalValue = FinalValue - curVal;
            }
        }
        return FinalValue;
    }
        
        public static int value(char num) {
            if(num == 'I'){
                return 1;
            }
            if(num == 'V'){
                return 5;
            }else if(num == 'X'){
                return 10;
            }else if(num == 'L'){
                return 50;
            }else if(num == 'C'){
                return 100;
            }else if(num == 'D'){
                return 500;
            }else if(num == 'M'){
                return 1000;
            }else {
                return -1;
            }
        }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        res = 0
        numLen = len(s)
        i=0
        hm ={"I":1,"V":5,"X":10,"L":50,"C":100,"D":500,"M":1000}
        while i<numLen:
            if s[i]=="I" and i<numLen-1:
                if s[i+1] == "V":
                    res = res+4
                    i+=2
                elif s[i+1] == "X":
                    res = res+9
                    i+=2
                else:
                    res = res+hm[s[i]]
                    i+=1
            elif s[i]=="X" and i<numLen-1:
                if s[i+1] == "L":
                    res = res+40
                    i+=2
                elif s[i+1] == "C":
                    res = res+90
                    i+=2
                else:
                    res = res+hm[s[i]]
                    i+=1
            elif s[i]=="C" and i<numLen-1:
                if s[i+1] == "D":
                    res = res+400
                    i+=2
                elif s[i+1] == "M":
                    res = res+900
                    i+=2
                else:
                    res = res+hm[s[i]]
                    i+=1
            else:
                res = res+hm[s[i]]
                i+=1
        return res